sunpy/map/maphelper.py

```diff
@@ -0,0 +1,200 @@
+    >>> my_coord = SkyCoord(0*u.arcsec, 0*u.arcsec, obstime="2017-08-01", observer = 'earth', frame=frames.Helioprojective)
+    >>> my_header = map.make_header(data, my_coord)
+    >>> my_map = map.Map(data, my_header)
+    """
+    if not isinstance(coordinate, (SkyCoord, frames.BaseCoordinateFrame)):
+        raise ValueError("coordinate needs to be an Astropy coordinate frame or SkyCoord instance.")
+    meta = _get_meta_from_coordinate(coordinate)
+
+    meta['crval1'], meta['crval2'] = coordinate.Tx.value, coordinate.Ty.value
+
+    if reference_pixel == 'center':
+        reference_pixel = u.Quantity([(data.shape[0] + 1)/2.*u.pix, (data.shape[1] + 1)/2.*u.pix])
+
+    meta['crpix1'], meta['crpix2'] = reference_pixel[0], reference_pixel[1]
+
+    meta['cdelt1'], meta['cdelt2'] = scale[0], scale[1]
+
+    meta_dict = MetaDict(meta)
+
+    for key in kwargs:
+        if key in acceptable_meta:
+            meta_dict[key] = kwargs[key]
+
+    return meta_dict
+
+
+def _get_meta_from_coordinate(coordinate):
+    """
+    Function to get WCS meta data from an `astropy.coordinates.SkyCoord` or frame using
+    `astropy.wcs.utils.celestial_frame_to_wcs`.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseFrame`
+        Coordinate object to pull metadata from.
+
+    Returns
+    -------
+    `dict`
+        Containing the meta information required for `sunpy.map.Map` header:
+            * ctype1, ctype2
+            * cunit1, cunit2
+            * hgln_obs, hglt_obs
+            * dsun_obs
+            * rsun_obs
+            * date_obs
+    """
+
+    coord_meta = {}
+
+    if not isinstance(coordinate, (SkyCoord, frames.BaseCoordinateFrame)):
+        raise ValueError("Input must be an Astropy coordinate frame or SkyCoord instance.")
+
+    if isinstance(coordinate, SkyCoord):
+        if coordinate.obstime is None or coordinate.frame is None:
+            raise ValueError("SkyCoord needs an observation time and a frame.")
+        skycoord_wcs = astropy.wcs.utils.celestial_frame_to_wcs(coordinate.frame)
+    elif isinstance(coordinate, frames.SunPyBaseCoordinateFrame):
+        if coordinate.obstime is None:
+            raise ValueError("Frame needs an observation time.")
+        skycoord_wcs = astropy.wcs.utils.celestial_frame_to_wcs(coordinate)
+
+    cunit1, cunit2 = skycoord_wcs.wcs.cunit
+    coord_meta = dict(skycoord_wcs.to_header())
+    coord_meta['cunit1'], coord_meta['cunit2'] = cunit1, cunit2  # this is done as to_header transforms cunit to deg
+    coord_meta['hgln_obs'], coord_meta['hglt_obs'] = skycoord_wcs.heliographic_observer.lon, skycoord_wcs.heliographic_observer.lat
+    coord_meta['dsun_obs'] = skycoord_wcs.heliographic_observer.radius.to(u.m)
+    coord_meta['rsun_ref'] = skycoord_wcs.rsun.to(u.m)
+    coord_meta['rsun_obs'] = ((skycoord_wcs.rsun.to(u.AU)/skycoord_wcs.heliographic_observer.radius.to(u.AU)).decompose() * u.radian).to(u.arcsec).value
```

Comment:
Is this equation correct? Shouldn't this be an angular quantity?

Also you don't need to to `.to(u.AU)` on both sides of this, the `Quantity` magic will handle all the required unit conversion for you.

Reply:
Good question - I'm actually not too sure?